install_script.sh.template

```diff
@@ -378,6 +378,214 @@ function ensure_config_file_exists() {
+                        "_sampling_priority_v1": 2
+                    }
+                }
+            ]]
+        }
+    }
+END
+)
+
+    printf -v log_lines_stderr "%b" "$install_stderr"
+    printf -v log_lines_stdout "%b" "$install_stdout"
+    json_logs="["
+    while IFS= read -r line; do
+        if [ -z "$line" ]; then
+            continue
+        fi
+        json_logs+="{\"message\": \"$line\", \"level\": \"ERROR\", \"trace_id\": \"${trace_id}\", \"span_id\": \"${trace_id}\"},"
+    done <<< "$log_lines_stderr"
+    while IFS= read -r line;
+    do
+        if [ -z "$line" ]; then
+            continue
+        fi
+        json_logs+="{\"message\": \"$line\", \"level\": \"DEBUG\", \"trace_id\": \"${trace_id}\", \"span_id\": \"${trace_id}\"},"
+    done <<< "$log_lines_stdout"
+    json_logs="${json_logs%,}]"
+
+    telemetry_logs=$(cat <<-END
+    {
+        "api_version": "v2",
+        "request_type": "logs",
+        "tracer_time": ${time_now_seconds},
+        "runtime_id": "${trace_id}",
+        "seq_id": 2,
+        "origin": "linux-install-script",
+        "host": {
+            "hostname": "$(uname -n)",
+            "os": "${os}",
+            "distribution": "${distribution}",
+            "architecture": "$(uname -m)",
+            "kernel_version": "$(uname -v)",
+            "kernel_name": "$(uname -s)",
+            "kernel_release": "$(uname -r)"
+        },
+        "application": {
+            "service_name": "datadog-linux-install-script",
+            "service_version": "${install_script_version}",
+            "language_name": "UNKNOWN",
+            "language_version": "n/a",
+            "tracer_version": "n/a"
+        },
+        "payload": {
+            "logs": ${json_logs}
+        }
+    }
+END
+)
+
+    if ! echo "$telemetry_logs" | curl -f -sSL --retry 5 -o /dev/null \
+        "$telemetry_url" \
+        --header 'Content-Type: application/json' \
+        --header "DD-Api-Key: $apikey" \
+        --data @-
+    then
+        echo "Unable to send logs telemetry\n"
+    fi
+
+    if ! echo "$telemetry_trace" | curl -f -sSL --retry 5 -o /dev/null \
+    "$telemetry_url" \
+    --header 'Content-Type: application/json' \
+    --header "DD-Api-Key: $apikey" \
+    --data @-
+    then
+        echo "Unable to send trace telemetry\n"
+    fi
+}
+
+function configure_installer() {
+    local sudo_cmd="$1"
+    local trace_id="$2"
+    local apm_instrumentation_enabled="$3"
+
+    $sudo_cmd mkdir -p /etc/datadog-agent
+    $sudo_cmd bash -c "cat <<EOT > /etc/datadog-agent/datadog-installer.ini
+[features]
+apm_instrumentation=$apm_instrumentation_enabled
+
+[telemetry]
+trace_id=$trace_id
+parent_id=$trace_id
+priority=2
+EOT"
+}
+
+function _install_installer() {
+    local sudo_cmd="$1"
+    local os="$2"
+    local distribution="$3"
+    local apm_instrumentation_enabled="$4"
+
+    if [ "$os" == "Debian" ] && [ "$distribution" == "Debian" ] && [ -n "$apm_instrumentation_enabled" ]; then
```

Comment:
`OS` is normalized to our handled distribution, you can probably remove the comparison on `$distribution`

Reply:
I initially wanted debian specifically and not ubuntu to limit as much as possible the scope but we might revisit. (AFAICT it's possible to have $os/$distribution be either debian/debian or debian/ubuntu if I read https://github.com/DataDog/agent-linux-install-script/blob/a65b0c07a6a8c11a105c13dfced1f326d8a10195/install_script.sh.template#L794-L795 properly)

Reply:
Ah ok gotcha, then all good